components/fleetArchiveDS.js

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { Datasource } = require("./datasource.js");
+const logging = require("./logging.js");
+const _ = require("lodash");
+
+class FleetArchiveLogs extends Datasource {
+  constructor(argv) {
+    super(argv);
+    this.startTimeSeconds = Math.round(
+      new Date(argv.startTime).getTime() / 1000
+    );
+    this.endTimeSeconds = Math.round(new Date(argv.endTime).getTime() / 1000);
+  }
+  async fetchTripLogsForVehicle(vehicle_id, vehicleLogs, jwt) {
```

Comment:
To match the corresponding method in Cloud Logging, this should actually also return a trip if there's no vehicle id assigned to it.

For example, this method actually returns both trip and vehicle logs https://github.com/googlemaps/fleet-debugger/blob/main/components/cloudLoggingDS.js#L93

If the fetchVehicleLogs method only returns `trip` logs, `fetchTripLogsForVehicle` can return only `trip` logs with no associated `vehicle` as well.

Reply:
@greghutch to validate this behavior

Reply:
I think I tried to match exactly what Cloud Logging did.

If a vehicle gets passed in,
- For fetchVehicleLogs, it would get both the vehicle and trip logs that have the given vehicle id
- Later in fetchTripLogsForVehicle, it would get all the trip logs for each trip the original vehicle was part of.

If a trip gets passed in,
- For fetchVehicleLogs, it would get both the vehicle and trip logs that have the given trip
- Later in fetchTripLogsForVehicle, it would return an **empty list** because the vehicle_id doesn't exist.